src/useBeforeUnload.ts

```diff
@@ -0,0 +1,17 @@
+import {useEffect} from "react";
+
+const useBeforeUnload = (message?: string) => {
+  useEffect(() => {
+    window.onbeforeunload = e => {
```

Comment:
Needs `event.preventDefault()`? https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#Examples

Also, are there any particular reasons to use this type of event handler over `window.addEventListener(...)`?

Reply:
Done